commit/merkleroot/rmn/streamconfig.go

```diff
@@ -0,0 +1,160 @@
+package rmn
+
+import (
+	"math"
+	"time"
+
+	"github.com/smartcontractkit/libocr/networking"
+	"github.com/smartcontractkit/libocr/ragep2p"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
+	"github.com/smartcontractkit/chainlink-ccip/commit/merkleroot/rmn/rmnpb"
+)
+
+const (
+	// estimatedMaxNumberOfSourceChains is the estimated maximum number of source chains
+	// that the current stream configuration supports and can be increased if required.
+	// This value does not correlate to the maximum number of source chains that CCIP can support.
+	estimatedMaxNumberOfSourceChains = 500
+
+	// initialObservationRequest + observationRequestWithOtherSourcesAfterTimeout + reportSignatureRequest
+	maxNumOfMsgsPerRound = 3
+
+	// values below chosen by research team
+	rateScale     = 1.2
+	capacityScale = 3
+
+	estimatedRoundInterval = time.Second
+)
+
+var (
+	maxObservationRequestBytes int
+	maxReportSigRequestBytes   int
+)
+
+func newStreamConfig(
+	lggr logger.Logger,
+	streamName string,
+) networking.NewStreamArgs1 {
+	cfg := networking.NewStreamArgs1{
+		StreamName:         streamName,
+		OutgoingBufferSize: 1, // should always be set to 1.
+		IncomingBufferSize: 1, // should always be set to 1.
```

Comment:
Could we add these magic numbers as a const? An explanation to why they should always be set to 1 would be helpful.